testsuite/cluster-test/src/experiments/load_test.rs

```diff
@@ -202,21 +236,24 @@ struct StubbedNode {
 }
 
 impl StubbedNode {
-    async fn launch(node_endpoint: String) -> Self {
+    async fn launch(node_endpoint: String, index: usize) -> Self {
         // generate seed peers config from querying node endpoint
         let seed_peers = seed_peer_generator::utils::gen_seed_peer_config(node_endpoint);
 
         // build sparse network runner
 
-        let pfn_config = NodeConfig::default_for_public_full_node();
+        let mut pfn_config = NodeConfig::default_for_public_full_node();
 
         // some sanity checks on the network the stubbed node will be running in
         assert_eq!(
             pfn_config.full_node_networks.len(),
             1,
             "expected only one fn network for PFN"
         );
-        let network_config = &pfn_config.full_node_networks[0];
+        let network_config = &mut pfn_config.full_node_networks[0];
+        network_config.listen_address = format!("/ip4/127.0.0.1/tcp/{}", 6180 + index)
+            .parse()
+            .unwrap();
```

Comment:
also at line 272, probably worth reusing the same runtime instead of creating one per stubbed.